### PR TITLE
Extending Environmental testing HS-1253

### PR DESCRIPTION
**Pull request by chuklov**
## Description
Adding new methods to remove nodes and to get node ID as well as list of nodes

## Jira link(s)
- https://issues.opennms.org/browse/HS-1253

### CODE REVIEW

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        return new URL(new URL(baseUrl), path);
+    }
+
+    /**
+     * Method to get the node ID from DB
+     * @return Node ID as Int
+     * @throws MalformedURLException
+     */
+    public int getNodeId() throws MalformedURLException {
```
Comment (artnaseef) [inline external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java:113]: How about naming this `getFirstNodeId()` to help make this "Real Obvious Code"?

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        return id;
+    }
+
+    /**
+     * Method to chec the status of the node during the test
+     * @param status Expected status of the node
+     * @return If the status is equals tot eh expected one
+     * @throws MalformedURLException
+     */
+    public boolean checkTheStatusOfTheNode(String status) throws MalformedURLException {
```
Comment (artnaseef) [inline external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java:137]: Ditton on this one - how about naming it `checkTheStatusOfTheFirstNode`?

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java [hunk 3]
```diff
@@ -145,6 +148,50 @@ public void addADeviceWithLabelIPAddressAndLocation(String label, String ipAddre
             ( createNodeResult.getErrors() == null ) || ( createNodeResult.getErrors().isEmpty() ));
     }
 
+
+    @Then("Check the status of the Node {string}")
+    public void checkTheStatusOfTheNodeTest(String status) throws MalformedURLException {
```
Comment (artnaseef) [inline external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java:153]: Ditto on the naming.

Also, the text is a little vague - it's unclear what the string means.  How about `Check the status of the Node = {string}` (adding `=`)?

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java [hunk 4]
```diff
@@ -145,6 +148,50 @@ public void addADeviceWithLabelIPAddressAndLocation(String label, String ipAddre
...
+            ;
+            assertTrue(true);
+        } catch (Exception e) {
+            LOG.info("Test status: ", e.getMessage());
+            assertTrue(false);
+        }
+    }
+
+    @Then("Delete created node from inventory")
+    public void deleteNodeFromInventory() throws MalformedURLException {
```
Comment (artnaseef) [inline external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java:170]: Ditto on the naming.

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java [hunk 5]
```diff
@@ -227,4 +274,57 @@ private List<MinionData> commonFilterMinionsAtLocation(FindAllMinionsQueryResult
...
+        String currentStatus = (String) map.get("status");
+        return currentStatus.equals(status);
+    }
+
+    /**
+     * Method to get the node ID from DB
+     * @return Node ID as Int
+     * @throws MalformedURLException
+     */
+    public int getNodeId() throws MalformedURLException {
```
Comment (artnaseef) [inline external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java:312]: Ditto on the naming.

File: external-it/external-horizon-stream-it/pom.xml [hunk 6]
```diff
@@ -109,6 +110,11 @@
             <version>4.1.1</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
```
Comment (artnaseef) [inline external-it/external-horizon-stream-it/pom.xml:113]: Are test containers used?

**Review by artnaseef — COMMENTED** [review]
Looks like a good next step.

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java (hunk 4 above)
Comment (chuklov) [reply external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java:170]: Changed to be more precise

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java [hunk 7]
```diff
@@ -227,4 +274,57 @@ private List<MinionData> commonFilterMinionsAtLocation(FindAllMinionsQueryResult
         return minionsAtLocation;
     }
 
+    /**
+     * Method to chec the status of the node during the test
+     * @param status Expected status of the node
+     * @return If the status is equals tot eh expected one
+     * @throws MalformedURLException
+     */
+    public boolean checkTheStatusOfTheNode(String status) throws MalformedURLException {
```
[the remark this answers is not in the document]
Comment (chuklov) [reply external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java:283]: Changed to be more precise

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java (hunk 5 above)
Comment (chuklov) [reply external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/InventoryTestSteps.java:312]: Changed to be more precise

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java (hunk 2 above)
Comment (chuklov) [reply external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java:137]: Changed to be more precise

File: external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java (hunk 1 above)
Comment (chuklov) [reply external-it/external-horizon-stream-it/src/main/java/org/opennms/horizon/it/DiscoveryTest.java:113]: Changed to be more precise

**Review by artnaseef — APPROVED** [review]

**Pull request merged**